unit-tests/live/frames/test-fps-permutations.py

```diff
@@ -0,0 +1,324 @@
+                s += f"Profile {profile.stream_name()} expects {profile.fps()} fps on {get_resolution(profile)}\n"
+        else:
+            s += (f"Expected fps for sensor {sensor_key.name} on profile "
+                  f"{sensor_profile_dict[sensor_key][0].stream_name()} is {expected_fps_dict[sensor_key]}"
+                  f" on {get_resolution(sensor_profile_dict[sensor_key][0])}\n")
+    s = s.replace("on (0, 0)", "")  # remove no resolution for Motion Module profiles
+    s += "***************"
+    return s
+
+
+def perform_fps_test(sensor_profiles_arr, modes):
+
+    log.d(get_time_est_string(len(sensor_profiles_arr), modes))
+
+    for mode in modes:
+        # Generate a list of all possible combinations of 1s and 0s (permutations) in the length of the array
+        perms = list(itertools.product([0, 1], repeat=len(sensor_profiles_arr)))
+
+        # Go over every possible permutation and check if we should test it
+        # Each index in a permutation represents a profile, which will be tested only if the value in that index is 1
+        for perm in perms:
+            # log.d(perm)
+            if all(v == 0 for v in perm):
+                continue
+            if should_test(mode, perm):
+                partial_dict = get_dict_for_permutation(sensor_profiles_arr, perm)
+                test.start("Testing", get_tested_profiles_string(partial_dict))
+                expected_fps = get_expected_fps(partial_dict)
+                log.d(get_test_details_str(partial_dict, expected_fps))
+                fps_dict = measure_fps(partial_dict)
+                test.check(check_fps(expected_fps, fps_dict))
+                test.finish()
+
+
+####################################################
+# ---------- Test Preparation Functions ---------- #
+####################################################
+def get_profiles_by_resolution(sensor, resolution, fps=None):
+    # resolution is a required parameter because on a sensor all active profiles must have the same resolution
+    profiles = []
+    p_streams_added = []
+    for p in sensor.get_stream_profiles():
+        if get_resolution(p) == resolution:
+            if fps is None or p.fps() == fps:
+                if p.stream_type() not in p_streams_added:  # can't open same stream twice
+                    profiles.append(p)
+                    p_streams_added.append(p.stream_type())
+    return profiles
+
+
+def get_mutual_resolution(sensor):
+    profile_resolutions_dict = {}  # a map between a stream type and all of its possible resolutions
+    possible_combinations = []
+    for profile in sensor.get_stream_profiles():
+        stream_type = profile.stream_type()
+        resolution = get_resolution(profile)
+        fps = profile.fps()
+
+        # d[key] = d.get(key, []) + [value] -> adds to the dictionary or appends if it exists
+        profile_resolutions_dict[stream_type] = profile_resolutions_dict.get(stream_type, []) + [resolution]
```

Comment:
IR1 and IR2 cannot be selected together (and colored IR where supported..).
This might be OK if we want to shorten the test time but it should be noted (comment)

Reply:
I'll add a comment mentioning we only take one from each stream under 'get_profiles_by_resolution'